src/workerd/io/worker-entrypoint.c++

```diff
@@ -498,6 +498,31 @@ kj::Promise<void> WorkerEntrypoint::connect(kj::StringPtr host,
     kj::AsyncIoStream& connection,
     ConnectResponse& response,
     kj::HttpConnectSettings settings) {
+  TRACE_EVENT("workerd", "WorkerEntrypoint::runScheduled()");
+  auto incomingRequest =
+      kj::mv(KJ_REQUIRE_NONNULL(this->incomingRequest, "connect() can only be called once"));
+  this->incomingRequest = kj::none;
+  incomingRequest->delivered();
+  auto& context = incomingRequest->getContext();
+
+  KJ_DEFER({
+    // Since we called incomingRequest->delivered, we are obliged to call `drain()`.
+    auto promise = incomingRequest->drain().attach(kj::mv(incomingRequest));
+    waitUntilTasks.add(maybeAddGcPassForTest(context, kj::mv(promise)));
+  });
+
+  if (context.getWorker().getIsolate().getApi().getFeatureFlags().getConnectPassThrough()) {
+    // connect_pass_through feature flag means we should just forward the connect request on to
+    // the global outbound.
+
+    auto next = context.getSubrequestChannelNoChecks(
+        IoContext::NEXT_CLIENT_CHANNEL, false, kj::mv(cfBlobJson));
+
+    // Note: Intentionally return without co_await so that the `incomingRequest` is destroyed,
+    //   because we don't have any need to keep the context around.
+    return next->connect(host, headers, connection, response, settings);
+  }
+
   JSG_FAIL_REQUIRE(TypeError, "Incoming CONNECT on a worker not supported");
```

Comment:
```suggestion
  JSG_REQUIRE(context.getWorker().getIsolate().getApi().getFeatureFlags().getConnectPassThrough(),
      TypeError, "Incoming CONNECT on a worker not supported");

  // connect_pass_through feature flag means we should just forward the connect request on to
  // the global outbound.

  auto next = context.getSubrequestChannelNoChecks(
      IoContext::NEXT_CLIENT_CHANNEL, false, kj::mv(cfBlobJson));

  // Note: Intentionally return without co_await so that the `incomingRequest` is destroyed,
  //   because we don't have any need to keep the context around.
  return next->connect(host, headers, connection, response, settings);
```

Reply:
I structured it this way intentionally as I assume we'll need to continue supporting this flag if and when we create a real API. When that happens we'll remove the assert at the end of the function and add code there, but not have to touch the block that tests the feature flag and handles it.